src/Interfaces/QuoteProgressValidator.php

```diff
@@ -0,0 +1,10 @@
+<?php
+
+namespace Jauntin\SavingQuote\Interfaces;
+
+interface QuoteProgressValidator
+{
+    public function rules(array $data): void;
```

Comment:
@DashaSyvyuk is this a setter? If it's a setter, then it would make sense to return void.

Is this a case of the AI being confused about the purpose of `rules`, not realizing it's a setter? Maybe it should be `setRules`? I would look at existing Laravel Validator https://laravel.com/api/10.x/Illuminate/Validation/Validator.html for either an interface to implement or at least just a template to make a custom interface.